core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/wordCloud/WordCloudOpFinalExec.java

```diff
@@ -38,42 +46,79 @@ public void close() {
         termFreqMap = null;
     }
 
-    @Override
-    public String getParam(String query) {
-        return null;
+    public List<Tuple> normalizeWordCloudTuples() {
+        double minValue = Double.MAX_VALUE;
+        double maxValue = Double.MIN_VALUE;
+
+        for (Map.Entry<String, Integer> e : termFreqMap.entrySet()) {
+            int frequency = e.getValue();
+            minValue = Math.min(minValue, frequency);
+            maxValue = Math.max(maxValue, frequency);
+        }
+        // normalize the font size for wordcloud js
+        // https://github.com/timdream/wordcloud2.js/issues/53
+        List<Tuple> termFreqTuples = new ArrayList<>();
+        for (Map.Entry<String, Integer> e : termFreqMap.entrySet()) {
+            termFreqTuples.add(Tuple.newBuilder().add(
+                    resultSchema,
+                    Arrays.asList(
+                            e.getKey(),
+                            (int) ((e.getValue() - minValue) / (maxValue - minValue) *
+                                    (MAX_FONT_SIZE - MIN_FONT_SIZE) + MIN_FONT_SIZE)
+                    )
+            ).build());
```

Comment:
my understanding is backend is only yielding `(word, count)` pair and possibly with `isInsert` flag to the frontend. I don't get what does the `font size` do in this result payload.

Reply:
this is removed after shifting to the new implementation